src/wasm-delegations-fields.h

```diff
@@ -0,0 +1,582 @@
+/*
+ * Copyright 2020 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Implements a switch on an expression class ID, and has a case for each id
+// in which it runs delegates on the fields and immediates. All the delegates
+// are optional, so you can just provide what you want, and no code will be
+// emitted for the others.
+//
+// (The only mandatory thing to define is DELEGATE_ID which is the key for the
+// switch.)
+//
+// All #defines used here are undefed automatically at the end for you.
+//
+// Child pointers are emitted in reverse order (which is convenient for walking
+// by pushing them to a stack first).
```

Comment:
Does this order matter? I don't think it matters in copying, but I guess you plan to use this in some other place that matters..?

Reply:
Yes, it will matter in walking, but does not matter yet, in copying.